### ISSUE TITLE
Let's talk about new features in plugins

### ISSUE THREAD

**Issue by Arkshine** [alliedmodders/amxmodx#609]
Before we go wild on plugins, we should discuss what changes/new features/new plugins would be welcomed for each plugin. It will be easier to manage people's ideas and we will get a better overview of the work to do. We can also sort them by priority and whether they are unwanted/out-of-scope/impossible/etc. Some changes may require modifications in core, therefore it's important to discuss everything beforehand.

I will try to update below based on the proposed changes.

_waiting for feedbacks_

* Admin
* Admin Chat
* Admin Cmds
* Admin Help
* Admin Votes
* Anti Flood
* Commands Menu
* Info. Messages
* Maps Menu
* Menus Front-End
* Multi-Lingual System
* Nextmap Chooser
* NextMap
* Pause Plugins
* Players Menu
* Plugin Menu
* Slots Reservation
* Scrolling Message
* Stats Configuration
* Teleport Menu
* TimeLeft

* CS Misc. Stats
* CS Stats Logging
* CS StatsX
* CS Restrict Weapons

**Comment by wopox1337** [comment]
**Slots Reservation**
 I think we need to update him functionality.
1) Add kick random(?) player when reserved slots is reached.
2) Give a some API for thir-party plugins, give some events like a check player before kick & another. 
3) Now plugin check only client flag ADMIN_RESERVATION on _client_authorized(id)_ . We can add another conditions for this.

**Comment by wopox1337** [comment]
**Nextmap Chooser**
Modern plugin to inform players represent a much greater functionality.
Just look at one of them.:
1. [MapManager [Modular]](https://github.com/Mistrick/MapManagerModular) by @Mistrick 
2. [Galileo v5.9.1-925](https://forums.alliedmods.net/showthread.php?t=273019)

**Comment by GoRiLliAz** [comment]
**GaG/UnGag**
As for CS: GO, embed the function Gag/UnGAG directly has AMX

**Comment by GoRiLliAz** [comment]
**Menu**
Automatically hidden the menu for which one has no access

**Comment by wopox1337** [comment]
**Admin**
Huh.... I don't even know where to start... Firstly, need to divide the management and loading privileges of players from the management of players (bans, etc.).
There should be a lot of work, because this plugin is always replaced by a custom one. I don't remember when I needed the standard =(.

**Comment by GoRiLliAz** [comment]
**Slap/slay**
Il faudrait pouvoir choisir si on souhaite mettre le slay en premier ou en dernier.

**Comment by GoRiLliAz** [comment]
**Team Balance**
The balancing system should be redone by improving it like this:
https://forums.alliedmods.net/showthread.php?t=26598

- With a change of team at the last second when a player is changed from team.
- As well as protection of players to avoid the abuse of kill at the end of the round.

**Comment by Arkshine** [comment]
> Add kick random(?) player when reserved slots is reached.

what do you mean?

**Comment by Arkshine** [comment]
Just giving existing plugin doesn't help much. We need a concrete list of features. It's kind of hard because all the plugins need to be overhauled. We need to keep compatibility too.

**Comment by GoRiLliAz** [comment]
**Slots Reservation**

Ajouter la fonction kick Spectateur ou AFK après un temps prédéfinis.

**Comment by GoRiLliAz** [comment]
@Arkshine Ok i edit my previous post later

**Comment by Arkshine** [comment]
> **Admin**
> Huh.... I don't even know where to start... Firstly, need to divide the management and loading privileges of players from the management of players (bans, etc.).
> There should be a lot of work, because this plugin is always replaced by a custom one. I don't remember when I needed the standard =(.

I wanted to do something similar to SM. We need an API for plugins too.

**Comment by wopox1337** [comment]
> > Add kick random(?) player when reserved slots is reached.
> 
> what do you mean?

@Arkshine  i mean this. if we have 2 reserved slots and this slots are busy. We should search player to kick from server. To free up space for a reserve slot.

**Comment by OciXCrom** [comment]
1. Colored chat messages in all plugins for Counter-Strike.

Since `client_print_color` uses `client_print` if the game isn't CS, it's perfectly save to implement color symbols in the default plugins without having to worry about other games.

The only change required in the core is modifying `client_print` to not show ^1, ^3 and ^4 - right now it doesn't actually show them, but instead it shows a huge gap whenever they are used in a message.

```pawn
client_print(id, print_chat, "This ^4is ^3a ^1test ^4message^1.")
```

Result:

![client_print](https://i.imgur.com/ZDP4oM5.png)

2. **admin.sma** - adding the ability to set an expiration date for admins. I started overhauling the plugin [here](https://github.com/alliedmodders/amxmodx/compare/master...OciXCrom:admin-overhaul), I'll try to implement this option when I'm done.

3. **nextmap.sma** - implementing the ability of [this](https://forums.alliedmods.net/showthread.php?p=188898) plugin as it is a very useful option to have. It can be implemented with a cvar to enable/disable it.

4. **adminvote.sma** - add a `amx_votecvar` command.

5. Use the new `AutoExecConfig` function to generate separate .cfg files for each plugin rather than having everything mashed up in `amxx.cfg`. I started a discussion in #577, but still no reply.

6. Maybe we can also add another plugin with extended admin commands from [AMX Super](https://forums.alliedmods.net/showthread.php?t=19542). Having new options for admins is always appreciated.

**Comment by WPMGPRoSToTeMa** [comment]
>     1. Add kick random(?) player when reserved slots is reached.

I think kicking most recent player can be default behavior.

**Comment by Arkshine** [comment]
> 1. Colored chat messages in all plugins for Counter-Strike.

I don't mind, but it should have a config for that, people may not like suddenly having color everywhere.
Also what messages are going to be changed, what colors to use. Ideally, I think people should decide how to color, but right now it's not really convenient. If you change the languages files, you might need to overwrite them later or you might want to change others languages. Probably not sanely doable though.

About symbol, amxx strips them from a language file, I guess this could be done for client_print_color.

> 1. **admin.sma** - adding the ability to set an expiration date for admins. I started overhauling the plugin [here](https://github.com/alliedmodders/amxmodx/compare/master...OciXCrom:admin-overhaul), I'll try to implement this option when I'm done.

Admin stuff needs to be a little more advanced, similar to SM, but yea, temporary admins has been requested several times.

> 4. Use the new `AutoExecConfig` function to generate separate .cfg files for each plugin rather than having everything mashed up in `amxx.cfg`. I started a discussion in #577, but still no reply.

I'm not sure. If we didn't have amxx.cfg with cvars already put inside, we could have used AutoExecConfig everywhere, but now it will be annoying for the user on upgrade if we move all the cvars related to the base plugins. Don't know, need more feedbacks on that.

> 5. Maybe we can also add another plugin with extended admin commands from [AMX Super](https://forums.alliedmods.net/showthread.php?t=19542), e.g. **funcmd.sma**. Having new options for admins is always appreciated.

I'm not against, but really not a priority, and you should list exactly what command you would want to include.

**Comment by dedimark** [comment]
Add Kickmenu with reasons.

**Comment by OciXCrom** [comment]
> I don't mind, but it should have a config for that, people may not like suddenly having color everywhere.
Also what messages are going to be changed, what colors to use. Ideally, I think people should decide how to color, but right now it's not really convenient. If you change the languages files, you might need to overwrite them later or you might want to change others languages. Probably not sanely doable though.

A cvar can be added to toggle the colors, it's not really a problem. Like you said, the only struggle will be adding the color symbols for all languages - but it's not something undoable. We should probably follow a simple rule when adding them, e.g. player names in team color and values in green, everything else in default color, so basically we need to add the colors before any `%s`, `%d`, etc. This way we won't have to worry about the actual words in other languages - they will retain the same color style.

I can do this if you make the `client_print` change. Oh, and yeah, the `show_activity` functions will need to be modified as well - #505.

> I'm not sure. If we didn't have amxx.cfg with cvars already put inside, we could have used AutoExecConfig everywhere, but now it will be annoying for the user on upgrade if we move all the cvars related to the base plugins. Don't know, need more feedbacks on that.

When you think about it, it's much more annoying as it is right now because with every new cvar added in AMXX, users will need to manually add that cvar in `amxx.cfg`. Take a look at your [release notes](https://wiki.alliedmods.net/AMX_Mod_X_1.9_Release_Notes#Config_changes) - these additional instructions can easily be avoided by using `AutoExecConfig` - new cvars will be automatically generated in the respective files and old ones will remain unchanged - users won't have to do the changes manually anymore. It's really the most appropriate thing to do IMO.

**Comment by voed** [comment]
I'd like to add natives to kick/ban player, using server_cmd triggering me every time

**Comment by justgo97** [comment]
I'm with adding AutoExecConfig for new cvars its frustrating to keep to track of these cvars or make amxx.cfg file to be generated only once like AutoExecConfig so that its not override every time a new cvar is being added.

**Comment by OciXCrom** [comment]
Yeah, we can also use `AutoExecConfig` to work with `amxx.cfg`, but it will require modifying the function because right now you can't use a `.cfg` file outside of the `configs/plugins` directory.

**Comment by Hembi** [comment]
Refereed to #581 :
- Add kick/ban reason to the kick and ban menu
- Add option to make a screenshot before the ban with an enable/disable cvar

- Add a reset score plugin with some command aliases
- AFK manager with some cvars (like afk time before do anything, transfer to spec or kick)
- Control cvar to enable/disable team flash which is compatible with HLDS and ReHLDS.

**Comment by CrazY770** [comment]
- Add a native to modify key value on textparse_ini.
- ArrayFindString with a case-sensitive param or ArrayFinsString2.
- amx_gag with flags (a = voice, b = say, c = say_team, etc)

**Comment by WPMGPRoSToTeMa** [comment]
> These are all mod-specific gameplay elements and should be altered with 3rd party plugins, just the way it has always been.

AFK kicking is a quite general, but it shouldn't be enabled by default like teleport. What about resetscore: looks like all GoldSrc games have some score systems, so it seems to be general too.

**Comment by luxxxoor** [comment]
Add amx_mute for voice chat.

**Comment by CrazY770** [comment]
> Add amx_mute for voice chat.

This is the same what I suggest above.

- amx_gag with flags (a = voice, b = say, c = say_team, etc)

**Comment by Maxi605** [comment]
> **Slots Reservation**
> I think we need to update him functionality.
> 
> 1. Add kick random(?) player when reserved slots is reached.
> 2. Give a some API for thir-party plugins, give some events like a check player before kick & another.
> 3. Now plugin check only client flag ADMIN_RESERVATION on _client_authorized(id)_ . We can add another conditions for this.

Make it as the server owner can choose to set on/off this feature (Feature 1), please, i use xRedirect for this specificly and i don't need players to be kicked, good idea btw.

**Comment by Maxi605** [comment]
Reasons for BanMenu should be a thing aswell, and maybe a option to tell the admin "Are you sure you want to ban %s?" (People may disagree with this and i'll understand why).

Restrict Weapons: Ability to limit the amount of a weapon that can be bought/in the world (Example: Limit 3 AWP's to be in use).

Slap/Slay: If the player doesn't have access to amx_slap or amx_slay make as the player can't use this options on the Slap/Slay Menu.

Votes: If the Gag/Mute gets implemented a new cvar amx_votemute "Player" "Time" "Reason" would be pretty good.

Stats: Knife Kills and HE Grande Kills would be good to have.

Scrolling Messages: The ability of giving messages only to 'X' person or Admins.

**Comment by HamletEagle** [comment]
> > **Slots Reservation**
> > I think we need to update him functionality.
> > 
> > 1. Add kick random(?) player when reserved slots is reached.
> > 2. Give a some API for thir-party plugins, give some events like a check player before kick & another.
> > 3. Now plugin check only client flag ADMIN_RESERVATION on _client_authorized(id)_ . We can add another conditions for this.
> 
> Make it as the server owner can choose to set on/off this feature (Feature 1), please, i use xRedirect for this specificly and i don't need players to be kicked, good idea btw.

You shouldn't be using plugins that work only on non steam clients/that are banned by valve.

**Comment by Maxi605** [comment]
> **Slots Reservation**
> 
> Add the function kick spectator or AFK after a preset time..

That's allready implemented in the game, mp_autokick, if set to 1 it will kick players after a certain time as AFK (In Spectator) and i think the players that teamkill and kill hostages.

**Comment by OciXCrom** [comment]
Restrict Weapons: restrict weapons by admin flags, i.e. ability to make specific weapons accessable only to people with a specific admin flag.

Admin Commands: show the kick reason in chat (toggleable via cvar).

**Comment by Maxi605** [comment]
`
] amx_addban
Usage:  amx_addban <"authid" or ip> <minutes> [reason]
] amx_addban "a" "a"
[AMXX] Authid "a" added to ban list`

Add a check on the "minutes" section of the command (all commands that need numbers) and don't allow to use any other key that's not a number, also add the need to have the reason filled (as it requires the name and time).

**Comment by Hembi** [comment]
Add earlier map vote and change by players option (rtv/rockthevote commands) to Nextmap Chooser

**Comment by Dr-Strange91** [comment]
amx_teammenu is general command and searching players from menu takes quite alot time.

A command should be added for it,

amx_team "steam id/nick/" "ct/ts/spec"

**Comment by rtxa** [comment]
> 
> 
> > **Slots Reservation**
> > Add the function kick spectator or AFK after a preset time..
> 
> That's allready implemented in the game, mp_autokick, if set to 1 it will kick players after a certain time as AFK (In Spectator) and i think the players that teamkill and kill hostages.

This is only implemented in CS 1.6, not in Half-Life for giving an example

**Comment by Maxi605** [comment]
> > > **Slots Reservation**
> > > Add the function kick spectator or AFK after a preset time..
> > 
> > 
> > That's allready implemented in the game, mp_autokick, if set to 1 it will kick players after a certain time as AFK (In Spectator) and i think the players that teamkill and kill hostages.
> 
> That is only implemented in CS 1.6, not in Half-Life for giving an example

Yeah, my bad, i keep thinking that AMXx is only for CS when it isn't

**Comment by Arkshine** [comment]
> * Add a native to modify key value and insert new data on textparse_ini.
> * ArrayFindString with a case-sensitive param or ArrayFindString2.

@CrazY770 you should make a separate issue for each of those. This is not related to plugins.

> Add a check on the "minutes" section of the command (all commands that need numbers) and don't allow to use any other key that's not a number, also add the need to have the reason filled (as it requires the name and time).

Yeah, it's really silly to not check argument validty.

> add Vote kick, ban, slay menu, kick ban slay authid or ip

@all85100 I'm not sure to understand that.

**Comment by Dr-Strange91** [comment]
Always Time Limit is being discussed in the core of AMXX Versions, why there ain't any focus on Round Limit?

**Comment by Dr-Strange91** [comment]
a seprate plugin is required for it, which is not in base.
I mean't to say All Extentions for map are over voting for time limit, not for Roundleft + mp_maxrounds, you need additional plugin.
If not, then my bad.

**Comment by Hembi** [comment]
I did not look after this, but what is the reason that plugins profiler did not implemented to the new version?
"https://forums.alliedmods.net/showpost.php?p=1527800&postcount=87" @Arkshine ?

**Comment by regener** [comment]
About the new AFK plugin, please add an option if game is cstrike to transfer bomb to another alive player if the AFK guy gets slayed/kicked.

**Comment by rsKliPPy** [comment]
That's why AFK manager shouldn't be in AMXX - you can't make everyone happy and it's **really** game-dependent. Just use a plugin.

**Comment by rtxa** [comment]
@rsKliPPy The same goes with menu for team change, in Half-Life there is a maximum of ten teams and it only let you change to CT, TT or Spectator

**Comment by gtteamamxx** [comment]
New plugin: Private messages for each player.
we have amx_psay, but this command can use only admin.
maybe we could extend this to `p@ nick_name/userid message`?
or.. add feature each player can pm each player.

**Comment by OciXCrom** [comment]
@gtteamamxx Why don't you simply remove the admin flag required for the command?

**Comment by gtteamamxx** [comment]
@OciXCrom We are talking about new features, not on "Why you don't gonna do it by yourself?"
I know i can do it, but we can allow user to configure this.. i don't know.. by cvar?
Adding possibility to write private message using chat, not console.

**Comment by Maxi605** [comment]
> New plugin: Private messages for each player.
> we have amx_psay, but this command can use only admin.
> maybe we could extend this to `p@ nick_name/userid message`?
> or.. add feature each player can pm each player.

It will open room for spamming on a server, not every server owner is experienced to even know that a flag can be changed and honestly i don't see a reason for this feature, if you want to message somebody, do it on steam.

**Comment by regener** [comment]
Please consider updating the cs_set_user_team function to work without crashing the server like this module: https://forums.alliedmods.net/showthread.php?t=163555

**Comment by Arkshine** [comment]
@regener It should be already fixed.

**Comment by Maxi605** [comment]
Add a cvar to allow admins to spectate everyone even if mp_forcecamera is set to 1

**Comment by Maxi605** [comment]
Ability to only let certain weapons be on certain maps, you can do it from the mapping side but a lot of mappers forget about this

**Comment by voed** [comment]
@Maxi605 use restmenu with map-related configs

**Comment by Maxi605** [comment]
> @Maxi605 use restmenu with map-related configs

Is it implemented on the restmenu the ability of limit items on maps with it?

**Comment by Dr-Strange91** [comment]
I would request here for a cvar in admin.amxx

L 11/28/2018 - 17:29:40: [admin.amxx] Login: "ABCD<3321><STEAM_0:0:1234567><>" became an admin (account "STEAM_0:0:1234567") (access "t") (address "39.36.21.333")

amx_cvar adminlog "on/off"

Sometimes so many logs hangs console due to VIP's.

**Comment by Maxi605** [comment]
by admin log you mean the conection of the admin?

i like it.

**Comment by Dr-Strange91** [comment]
> by admin log you mean the conection of the admin?
> 
> i like it.

I don't mean to remove it, I asked to put cvar to either keep it on or off !!

**Comment by Arkshine** [comment]
I'm not against, but I think it's quite important you can keep track of who is connecting with what IP/SteamID. I would prefer an option to log them in a separate file.

**Comment by Arkshine** [comment]
Oh, right, you meant still written in the log but not in the console.

**Comment by Dr-Strange91** [comment]
> Oh, right, you meant still written in the log but not in the console.

Yes !!!

**Comment by Maxi605** [comment]
> 1. When Server is restarted from Game Panel csstats are reset, so, I believe, fvault or nvault should be used in replacement of it.

what? is this true? can't this be fixed? D:

**Comment by Dr-Strange91** [comment]
Yes its true, you can check by yourself !!!

**Comment by Dr-Strange91** [comment]
I forget to mention, but I believe, a reason should be added with Kickmenu, like Banmenu.

amx_kickmenu

Type Reason after selecting player
Or Provide few Options, via lang/kickmenu.txt

**Comment by GoRiLliAz** [comment]
amx_ban
That the temporary ban is saved, because for the moment if we reboot the server the ban is removed
Which makes the temporary ban of more than 24 hours useless
amx_addban 
Show the duration of the ban in the chat.
For the moment there is just:
`ADMIN ~ | ShK | ~ GoRiLliAz.13: banned STEAM_0: 1: xxxxxxxx`
amx_addban
Hide the ip in the chat during a ban by ip, to be in the legality

**Comment by regener** [comment]
I don't restart the server every day, so I think that isn't a necessary change.
About the IP part: you can change the code in the admincmd.sma if you want.

**Comment by GoRiLliAz** [comment]
A CS 1.6 Server must be reboot every day... You should do it...

**Comment by WPMGPRoSToTeMa** [comment]
@GoRiLliAz why?

**Comment by GoRiLliAz** [comment]
Car le jeu gère mal la memoire, coté serveur, il est vite saturé.
Au bout de 24h, le serveur commence déjà à baisser en fluidité.

**Comment by regener** [comment]
That sounds like memory leak. Isn't it better to check your plugins and test stuff to avoid it, than restart it with still having the problem?

**Comment by GoRiLliAz** [comment]
^^ No memory leak ^^
90% of my plugins were made by Arkshine only for my team in 2015 and 2016.
I have out amxmodx 20 plugins 90% code by arkshine and the rest are plugins that are always updated on the forum alliedsmodder.
I urge you to reboot your server every 24h ...
Let's move on. Arkshine will decide if my proposal is good or not.

**Comment by Arkshine** [comment]
> That sounds like memory leak. Isn't it better to check your plugins and test stuff to avoid it, than restart it with still having the problem?

So far I remember HLDS is well known for leaking memory. It's old, bugged and unmaintained. You can check with Valgrind. Also, if you have a lot of activity/plugins and alter often the game, you will accumulate garbage and HLDS can do weird things at some point. Admins are not necessarily coders, and you can do nothing about HLDS, it's just easier to start from a clean state when you can.

**Comment by Dr-Strange91** [comment]
And what about servers like Death Match?

I think, Cvar should be added !!!

roundendtime "1/0" (1- New Behaviour)(0- Old Behaviour)?

NextMaps	- Allow the last round to be finished even if the timelimit has expired

**Comment by Hembi** [comment]
- Add Map specific users.ini like plugins.ini
(For example a MultiMod server with some VIP function this would be helpful)

**Comment by regener** [comment]
CS StatsX with kill assists and SQL db?
(I know it exists as a 3rd party, just asking)

**Comment by 4D1G06** [comment]
I have a request for the Albanian language support.
Not saying this just randomly, but based on the high number of servers who need this.
I can actually provide translations without any problem at all. I have did so in the past, but unfortunately lost them all.
This maybe is not the right place to make requests. If so, I sincerely apologize.
My regards, **thEsp** !

**Comment by OciXCrom** [comment]
@4D1G06 There's already a pull request for that language - https://github.com/alliedmodders/amxmodx/pulls

If you want to help, you can contribute there.

**Comment by regener** [comment]
Plugin Menu - https://github.com/alliedmodders/amxmodx/issues/735#issuecomment-499097205

**Comment by Dr-Strange91** [comment]
Are there any plans to add write Kick Reason, in amx_kickmenu.

**Comment by DarthMan** [comment]
I guess, this can still be considered a request for AMXX v1.10 and a good one. It's about the Orpheu module. I think it would be cool if such module is improved, with Vector support added like Okapi. And, ofc, it beeds to be included in AMXX v1.10 .  This way will make it possible to hook functions that are otherwise only possible yo hook with Okapi. Since Orpheu reads from files rather than havibg the sigs/offsets hardcoded, I think it's better if Irpheu is included instead. It also makes it easier to hook functions like pfnKeyValue for example. It's not too late. Maybe CS and the other games are dead but, at least, let us, those who still code for such dead games to be able to make amazing things without having to use other additional modules. I really hate how Okapi is hardcoded and if u want to support multiple mods for each new 1 u have to edit the plug-in again and again. With Orpheu it is much easier, allowing u to make a json-parse file without having to edit anything from the plug-in.

**Comment by voed** [comment]
@DarthMan this topic is about plugins, not modules 
And Orpheu is not even a part of AMXX

**Comment by DarthMan** [comment]
Yeah it is not, vut I requested that it comes with the default amxx package. And I seee no other place to post it on.

**Comment by GasmoN** [comment]
**Nextmap Chooser**
Don't include maps that were already played. (Only if there is X (cvar) new maps to choose from).
Reason: It happends that lot of players randomly press buttons or just like to play the same map over and over again, so it's kind of boring for others and they just leave. It's not a big deal when admins are online but when they are not it's a problem.

**Comment by djearthquake** [comment]
Port **AdminMod Map-Nomination-Vote Plugin** into core for rockethevote and nominate.

**Comment by Karaulov** [comment]
Add debug to amxx! Example https://github.com/alliedmodders/amxmodx/issues/794 :(

**Comment by SmileYzn** [comment]
Avoid usage of setinfo for terrible _pw, i think some users reported that can be stolen (If really can!)

LOL

**Comment by ma-taosatefuta** [comment]
Multi-language system

- Autotranslate words e.g. from any language to any other language based on player's language / server's in chat / menus and hudmessages

**Comment by DeclineD2CS** [comment]
> Avoid usage of setinfo for terrible _pw, i think some users reported that can be stolen (If really can!)
> 
> LOL

Totally, we can create plugins where we can use ge_user_info to get the passwords of any player we see and ask him if he's an admin on server then make him lose his admin and that's kinda bad.
But there's no way to solve it.

**Comment by DeclineD2CS** [comment]
Admin Help
- Ability to show all commands at ince or to set how many command you want to see on page and also the pages should be from 1 to infinity not 10, 20, 30, 40.

**Comment by OciXCrom** [comment]
> Multi-language system
> 
> * Autotranslate words e.g. from any language to any other language based on player's language / server's language(if the text isn't already in that one) in chat / menus and hudmessages

That would be chaotic and totally unfunctional. No auto-translating system is 100% accurate and players rarely write properly in chat, which will make the translation even that more unusable.

Menus and other plugin stuff should be translated manually.

**Comment by OciXCrom** [comment]
> Admin Help
> 
> * Ability to show all commands at once or to set how many commands you want to see on page and also the pages should be from 1 to infinity not 10, 20, 30, 40.

Having that many messages printed in the console at once can get the client kicked out from the server. That limitation is there for a reason.

**Comment by TibyXD** [comment]
The ability of having
amx_ct <player>
amx_t <player>
amx_spec <player>

Also, admin menu in chat, like /slay /slap /kick.

**Comment by OciXCrom** [comment]
@TibyXD those commands are redundant when you have `amx_teammenu`. If something like that is going to be added, it should be `amx_team` or `amx_transfer`. Having hardcoded commands only for CS is not ideal.

**Comment by wxzasd** [comment]
painshock_value (movement slowdown value when the player get hit) 
pushangle_value  (weapon/crosshair deviation when the player get hit)

**Comment by all85100** [comment]
Six years have passed, and none of the new features mentioned here have been implemented. Should everyone give up?

**Comment by dvander** [comment]
AMX Mod X is not actively maintained by anyone anymore. It is over 20 years old. If you're interested in taking over, feel free to submit some pull requests.-DavidOn Apr 16, 2024 5:23 AM, all85100 ***@***.***> wrote:
Six years have passed, and none of the new features mentioned here have been implemented. Should everyone give up?

**Comment by Arkshine** [comment]
I'm sorry, guys. The lack of general interest and IRL catching me up did not help me focus on the AMXX project. If I have the opportunity to come back, I will, but right now, I can't. Hopefully, someone can take over and continue working on it. 🙏

**Comment by DarthMan** [comment]
At least make 1.9 the default stable build, replacing 1.8.2 which is quite old.

**Comment by all85100** [comment]
> I'm sorry, guys. The lack of general interest and IRL catching me up did not help me focus on the AMXX project. If I have the opportunity to come back, I will, but right now, I can't. Hopefully, someone can take over and continue working on it. 🙏

I think project managers still need to take the time to deal with the right and reasonable Pull requests,
